### PR TITLE
[CFM-478] Send GO/GA notification on membership request, moderated group.

### PR DESCRIPTION
**Pull request by anvmn**
## This PR is dependent on #1059 

https://issuetracker.amplexor.com/jira/browse/CFM-478

### CODE REVIEW

**Comment by anvmn** [comment]
selection_114

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 1]
```diff
@@ -12,9 +12,40 @@ function c4m_user_notifications_notify_admins_user_deleted($account) {
...
+    // If message was not created properly, return early.
+    return;
+  }
+  c4m_message_load_tokens__c4m_new_membership_request($message);
+
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $wrapper->field_group_node->set($membership->gid);
+  $wrapper->field_new_member->set($membership->etid);
+  $wrapper->field_group_membership->set($membership->id);
+  message_notify_send_message($wrapper->value());
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: `message_subscribe_send_message()`

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -76,6 +77,24 @@ function _c4m_message_format_welcome_to_group_from_field(&$message, $message_ent
 }
 
 /**
+ * Formats 'c4m_new_membership_request' message From field, per requirement.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: @anvmn 
Can you please write the ID of the template ?
thanks

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -76,6 +77,24 @@ function _c4m_message_format_welcome_to_group_from_field(&$message, $message_ent
...
+  if ($message_entity->type != 'c4m_new_membership_request') {
+    // For message type other than c4m_new_membership_request - return early.
+    return;
+  }
+
+  $message['from'] = format_string(
+    '"The @site team"<@mail>',
+    array(
+      '@site' => variable_get('site_name'),
+      '@mail' => 'capacity4dev@ec.europa.eu',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: I think this should come from the database and not hardcoded
@jover @soniCaH @mfernea @janlemmens 
Can some of you confirm ?
and if so then where from ?

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: @soniCaH ^^

Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: There're many hard coded mail values at template document.
See MT48, for example.

**Comment by soniCaH** [comment]
@anvmn when testing the first issue that was merged in (CFM-454), there seems to be some functionality broken:

membershipreq

**Comment by anvmn** [comment]
## Replaced by #1189

**Pull request closed without merging**